google/cloud/storage/internal/grpc/metrics_exporter_impl_test.cc

```diff
@@ -0,0 +1,119 @@
+}
+
+auto FullResource() {
+  return opentelemetry::sdk::resource::Resource::Create(
+      {{"cloud.provider", "gcp"}, {"cloud.account.id", "project-id-resource"}});
+}
+
+auto TestOptions(Options opts = {}) {
+  opts.set<UnifiedCredentialsOption>(MakeAccessTokenCredentials(
+      "unused", std::chrono::system_clock::now() + std::chrono::minutes(15)));
+  return DefaultOptionsGrpc(std::move(opts));
+}
+
+TEST(GrpcMetricsExporter, DisabledBecauseOption) {
+  auto config = MakeMeterProviderConfig(
+      FullResource(),
+      TestOptions().set<storage_experimental::EnableGrpcMetricsOption>(false));
+  EXPECT_FALSE(config.has_value());
+}
+
+TEST(GrpcMetricsExporter, DisabledBecauseNoProject) {
+  auto config = MakeMeterProviderConfig(EmptyResource(), TestOptions());
+}
+
+TEST(GrpcMetricsExporter, EnabledResourceProject) {
+  auto config = MakeMeterProviderConfig(FullResource(), TestOptions());
+  ASSERT_TRUE(config.has_value());
+  EXPECT_EQ(config->project, Project("project-id-resource"));
+  EXPECT_TRUE(
+      config->exporter_options.has<otel_internal::MonitoredResourceOption>());
+  EXPECT_TRUE(
+      config->exporter_options.has<otel_internal::MetricPrefixOption>());
+  EXPECT_TRUE(
+      config->exporter_options.get<otel_internal::ServiceTimeSeriesOption>());
+  EXPECT_GT(config->reader_options.export_interval_millis,
+            std::chrono::milliseconds(0));
+  EXPECT_GT(config->reader_options.export_timeout_millis,
+            std::chrono::milliseconds(0));
+}
+
+TEST(GrpcMetricsExporter, EnabledOptionsProject) {
+  auto config = MakeMeterProviderConfig(
+      EmptyResource(),
+      TestOptions().set<storage::ProjectIdOption>("project-id-option"));
+  ASSERT_TRUE(config.has_value());
+  EXPECT_EQ(config->project, Project("project-id-option"));
+  EXPECT_TRUE(
+      config->exporter_options.has<otel_internal::MonitoredResourceOption>());
+  EXPECT_TRUE(
+      config->exporter_options.has<otel_internal::MetricPrefixOption>());
+  EXPECT_TRUE(
+      config->exporter_options.get<otel_internal::ServiceTimeSeriesOption>());
+  EXPECT_GT(config->reader_options.export_interval_millis,
+            std::chrono::milliseconds(0));
+  EXPECT_GT(config->reader_options.export_timeout_millis,
+            std::chrono::milliseconds(0));
+}
+
+TEST(GrpcMetricsExporter, EnabledWithTimeout) {
+  auto config = MakeMeterProviderConfig(
+      FullResource(),
+      TestOptions().set<storage_experimental::GrpcMetricsPeriodOption>(
+          std::chrono::seconds(60)));
+  ASSERT_TRUE(config.has_value());
+  EXPECT_EQ(config->project, Project("project-id-resource"));
+  EXPECT_TRUE(
+      config->exporter_options.has<otel_internal::MonitoredResourceOption>());
+  EXPECT_TRUE(
+      config->exporter_options.has<otel_internal::MetricPrefixOption>());
+  EXPECT_TRUE(
+      config->exporter_options.get<otel_internal::ServiceTimeSeriesOption>());
+  EXPECT_GT(config->reader_options.export_interval_millis,
+            std::chrono::milliseconds(60));
```

Comment:
`EXPECT_EQ(..., std::chrono::seconds(60))`

Reply:
optional nit: For this test, I would only set an expectation on the timeout, to make it super-duper obvious what is under test.

But we have discussed this sort of thing before, and I think we just have different preferences.